Common/KeyMap.cpp

```diff
@@ -318,9 +323,9 @@ void UpdateConfirmCancelKeys() {
 			confirmKeys.push_back(hardcodedConfirmKeys[i]);
 	}
 
-	const keycode_t hardcodedCancelKeys[] = {
-		NKCODE_ESCAPE,
-		NKCODE_BACK,
+	const KeyDef hardcodedCancelKeys[] = {
+		KeyDef(DEVICE_ID_KEYBOARD, NKCODE_ESCAPE),
+		KeyDef(DEVICE_ID_KEYBOARD, NKCODE_BACK),
```

Comment:
Referring to the NKCODE_BACK entry specifically, or both? One possible solution would be to add a special DEVICE_ID_ANY deviceId to signify "don't check deviceId", and special-case it in the comparison functions.